### PR TITLE
[@mantine/hooks] use-hover: Add tests

### PR DESCRIPTION
**Pull request by mctessi**
Hi!
I created some simple tests to verify the use-hover functionality! 
Every feedback is deeply appreciated and there are other test files waiting to be commited! 
Happy coding

### CODE REVIEW

File: src/mantine-hooks/src/use-hover/use-hover.test.tsx [hunk 1]
```diff
@@ -0,0 +1,27 @@
...
+      {hovered ? 'true' : 'false'}
+    </div>
+  );
+};
+
+describe('@mantine/hooks/use-hover', () => {
+  render(<Target />);
+  const target = screen.getByTestId('target');
+  it('changes "hovered" on mouseenter/mouseleave correctly', () => {
+    expect(target).toHaveTextContent(/false/i);
```
Comment (rtivital) [inline src/mantine-hooks/src/use-hover/use-hover.test.tsx:19]: regex is not required, use strings instead.

File: src/mantine-hooks/src/use-hover/use-hover.test.tsx [hunk 2]
```diff
@@ -0,0 +1,27 @@
...
+
+  return (
+    <div data-testid="target" ref={ref}>
+      {hovered ? 'true' : 'false'}
+    </div>
+  );
+};
+
+describe('@mantine/hooks/use-hover', () => {
+  render(<Target />);
```
Comment (rtivital) [inline src/mantine-hooks/src/use-hover/use-hover.test.tsx:16]: render inside `it` function call instead.

File: src/mantine-hooks/src/use-hover/use-hover.test.tsx (hunk 1 above)
Comment (mctessi) [reply src/mantine-hooks/src/use-hover/use-hover.test.tsx]: Thanks for the feedback! I have updated my request...

File: src/mantine-hooks/src/use-hover/use-hover.test.tsx (hunk 2 above)
Comment (mctessi) [reply src/mantine-hooks/src/use-hover/use-hover.test.tsx]: Thanks for the feedback! I have updated my request...

**Pull request closed without merging**